### PR TITLE
Bookmark service

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

WIP!
Adding bookmark service to fusion workspace
Will apply to the following;
Grid
Garden
View

Filter and sidesheet will come later

Completes: AB#FILL_IN_YOUR_ISSUE_ID

## Type of change

Remove options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Checklist

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code.
-   [ ] I have linked my DevOps task using the AB# tag.
-   [x] My code is easy to read.
-   [x] Documentation and comments is added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: packages/workspace-core/src/lib/classes/bookmarkService/bookmarkService.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+export class BookmarkService<TBookmarkState extends Record<PropertyKey, unknown>> {
```
Comment (Noggling) [inline packages/workspace-core/src/lib/classes/bookmarkService/bookmarkService.ts:1]: Documentasjon

File: packages/workspace-fusion/src/lib/utils/dataSource/addDataSource.ts [hunk 2]
```diff
@@ -1,5 +1,5 @@
 import { DataSourceController } from '@workspace/data-source';
-import { DataFetchAsync, FusionMediator } from '../types';
+import { DataFetchAsync, FusionMediator } from '../..';
```
Comment (Noggling) [inline packages/workspace-fusion/src/lib/utils/dataSource/addDataSource.ts:2]: ikke bruk ...

Comment (Gustav-Eikaas) [inline packages/workspace-fusion/src/lib/utils/dataSource/addDataSource.ts:2]: fix import

File: packages/workspace-fusion/src/lib/utils/garden/configureDataChange.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { GardenController } from '@equinor/garden';
+import { FusionMediator } from '../../types';
+
+export function configureDataChange<TData, TError, TCustomGroupByKeys, TCustomState, TContext>(
```
Comment (Noggling) [inline packages/workspace-fusion/src/lib/utils/garden/configureDataChange.ts:4]: doc

File: packages/workspace-fusion/src/lib/utils/grid/addGrid.tsx [hunk 4]
```diff
@@ -25,7 +27,25 @@ export function addGrid<TData, TError>(
 	});
 }
 
-export function gridControllerBinder<TData, TError>(
+function onBookmark<TData, TError>(gridController: GridController<TData>, mediator: FusionMediator<TData, TError>) {
```
Comment (Noggling) [inline packages/workspace-fusion/src/lib/utils/grid/addGrid.tsx:30]: extract to file

File: packages/workspace-fusion/src/lib/utils/indexedDb/addIndexedDb.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { FusionUserSettings } from '../../classes/fusionUserSettings';
+import { FusionMediator } from '../../types';
+
+export async function addIndexedDb<TData, TError>(mediator: FusionMediator<TData, TError>) {
+	const db = new FusionUserSettings<TData>();
+	mediator.bookmarkService.onCapture((res) => db.save(res));
+
```
Comment (Noggling) [inline packages/workspace-fusion/src/lib/utils/indexedDb/addIndexedDb.ts:7]: @Noggling Lag us for setting storing in API

File: packages/ag-grid/src/lib/utils/columnState.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ */
+function listenForColumnResize(gridReady: GridReadyEvent, catchColumnState: () => void) {
+	let timeoutId;
+	gridReady.api.addEventListener('columnResized', () => {
+		if (timeoutId) {
+			console.log('event queued, waiting');
+			return;
+		}
+		timeoutId = setTimeout(() => {
+			console.log('Capturing column state');
```
Comment (Gustav-Eikaas) [inline packages/ag-grid/src/lib/utils/columnState.ts:41]: remove console.log

**Review by Noggling — APPROVED** [review]

**Pull request merged**